channels/generic/http.py

```diff
@@ -87,3 +69,28 @@ class AsyncHttpConsumer(AsyncConsumer):
         from here.
         """
         pass
+
+    async def http_request(self, message):
+        """
+        Async entrypoint - concatenates body fragments and hands off control
+        to ``self.handle`` when the body has been completely received.
+        """
+
+        try:
+            if "body" in message:
+                self.body.append(message["body"])
+            if not message.get("more_body"):
+                await self.handle(b"".join(self.body))
+
+                await self.disconnect()
+                raise StopConsumer()
+        except Exception:
```

Comment:
```suggestion
        finally:
```

Reply:
@seroy Won't this `finally` be still executed even if there is more body to receive from the request? We only want to disconnect in case there was an error or it has been completed, in the latter case, it could take several events to happen.

Reply:
oh, yep! in this case:
```
async def http_request(self, message):
    """
    Async entrypoint - concatenates body fragments and hands off control
    to ``self.handle`` when the body has been completely received.
    """
    if "body" in message:
        self.body.append(message["body"])
    if not message.get("more_body"):
        try:
            await self.handle(b"".join(self.body))
        finnaly:
            await self.disconnect()
            raise StopConsumer()
```
Anyway catching Exception almost always is a bad Idea...

Reply:
@seroy Sorry for making you write the code lol, but thanks!